packages/mds-api-server/middleware/request-logging.ts

```diff
@@ -19,13 +19,14 @@ export const RequestLoggingMiddleware = <AccessTokenScope extends string>(
         '-',
         tokens['response-time'](req, res),
         'ms'
-      ].join(' ')
+      ]
+        .filter((token): token is string => token !== undefined)
```

Comment:
Looks pretty clean, I'd be curious to hear your reasoning behind making this filter a typeguard specifically beyond just simply returning a `boolean`.

Reply:
The typeguard is just to ensure correct type inference for subsequent methods; It's probably totally unnecessary, but at compile time without the typeguard you get `Array<string | undefined>` for the `.join`:

<img width="740" alt="Screen Shot 2020-07-23 at 10 23 14 PM" src="https://user-images.githubusercontent.com/3439869/88355635-4ef61e00-cd33-11ea-94a4-b8d5a907dc70.png">

Whereas _with_ the typeguard syntax you get `Array<string>`:

<img width="726" alt="Screen Shot 2020-07-23 at 10 22 47 PM" src="https://user-images.githubusercontent.com/3439869/88355686-79e07200-cd33-11ea-8ee4-e49af6aa87ed.png">

Without typeguards, `Array.filter` will never narrow the type of the array.